test/cli_test.py

```diff
@@ -350,6 +371,175 @@ def test_unkown_collection(self):
             self.runcli("alt", "update", "unkown")
         assert str(e.value) == "Alternative collection 'unkown' not found."
 
+    def test_update_all(self, tmp_path: Path):
+        dir_a = tmp_path / "a"
+        dir_a.mkdir()
+        dir_b = tmp_path / "b"
+        dir_b.mkdir()
+        self.config["alternatives"].get().clear()  # type: ignore
+        self.config["alternatives"] = {
+            "a": {
+                "directory": str(dir_a),
+                "query": "myexternal:true",
+            },
+            "b": {
+                "directory": str(dir_b),
+                "query": "myexternal:true",
+            },
+        }
+
+        with pytest.raises(UserError) as e:
+            self.runcli("alt", "update")
+        assert str(e.value) == "Please specify a collection name or the --all flag"
+
+        item = self.add_track(title="a", myexternal="true")
+        self.runcli("alt", "update", "--all")
+        item.load()
+        path_a = self.get_path(item, path_key="alt.a")
+        assert path_a
+        assert dir_a in path_a.parents
+        assert path_a.is_file()
+
+        path_b = self.get_path(item, path_key="alt.b")
+        assert path_b
+        assert dir_b in path_b.parents
+        assert path_b.is_file()
+
+        # Don’t update files on second run
+        assert self.runcli("alt", "update", "--all") == ""
+
+
+class TestExternalArt(TestHelper):
+    @pytest.fixture(autouse=True)
+    def _external_art(self, tmp_path: Path, _setup: None):
+        external_dir = str(tmp_path)
+        self.config["convert"]["embed"] = False
+        self.config["alternatives"] = {
+            "myexternal": {
+                "directory": external_dir,
+                "query": "myexternal:true",
+                "formats": "mp3",
+                "album_art_copy": False,
+                "album_art_maxwidth": None,
+                "album_art_format": None,
+                "album_art_deinterlace": False,
+                "album_art_quality": 0,
+            }
+        }
+        self.external_config = self.config["alternatives"]["myexternal"]
+
+    def touch_art(self, artpath: bytes, image_path: Path):
```

Comment:
Please keep this as a top-level function. I would like to avoid methods that don’t access the instance.